### ISSUE TITLE
Unable to obtain target process exit code on Linux/Mac with CoreCLR debugging services

### ISSUE THREAD

**Issue by gregg-miskelly** [dotnet/runtime#43731]
### Description

On Linux/Mac the exit code of the target process is obtained through waitpid ([Linux docs](https://linux.die.net/man/2/waitpid), [Apple docs](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/waitpid.2.html), [PAL code that makes use of it](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/pal/src/synchmgr/synchmanager.cpp#L4482)). However, unlike on Windows -- if multiple places in a process are calling waitpid, unless the `WNOWAIT` flag is used (which doesn't seem to exist OSX), then my understanding is that this will prevent other callers from obtaining the exit code.

The debugging services open a HANDLE to the target process (as expected) through the PAL, which means that they may wind up 'owning' the exit code of the target process, but they don't, at least as far as I know, wind up propagating this exit code to the debugger.

The potential fixes I can think of are:
1. Add a new 'managed callback' interface with another version of the [ExitProcess](https://github.com/dotnet/runtime/blob/c6bf8eb835b58c6bbbf1dc163ac026e97a9afdb1/src/coreclr/src/inc/cordebug.idl#L1121) method which provides the exit code to the debugger if the debugging services were able to obtain it.
-or-
2. Make changes to the PAL (ex: new fake access right to indicate that waitpid should be used) and opt-into them with the debugging services to avoid 'owning' the exit code. Thus the calling debugger can reliably use waitpid to obtain the exit code.

### Regression?

No, this has never worked on Linux/Mac

**Comment by pavel-orekhov** [comment]
same problem was in netcoredbg
cc @kfrolov @viewizard

**Comment by viewizard** [comment]
Yes, we also faced with this issue and temporary added hook for `waitpid` in netcoredbg code - https://github.com/Samsung/netcoredbg/blob/0809b8aa1a80c88b35a04106fbfeb9f5be42853e/src/debugger/manageddebugger.cpp#L41-L127
